src/main/java/com/cosium/code/format/InstallHooksMojo.java

```diff
@@ -74,6 +74,13 @@
   @Parameter(property = "gcf.preCommitHookPipeline", defaultValue = "")
   private String preCommitHookPipeline;
 
+  /**
+   * Set to `true` to use the global `mvn` executable in the hooks script otherwise will default to
+   * the complete path for `mvn` found via `maven.home` variable
+   */
+  @Parameter(property = "gcf.useGlobalMavenExecutable", defaultValue = "false")
+  private boolean useGlobalMavenExecutable;
```

Comment:
Are you using a maven wrapper? If not, what are you using exactly?
Isn't there some environment variable or anything set by the executed process to find its path?
That would allow us to avoid setting this property and instead add an automatic path detection in Maven Environment.

Reply:
I am not using a maven wrapper. I am using IntelliJ IDEA to run maven tasks.

It does infact set the `maven.home` property when any maven command is run via the IDE but the plugin somehow does not consider it.

IntelliJ sets the following property when running any maven command:

` -Dmaven.home=/Applications/JetBrains/apps/IDEA-U/ch-0/211.7442.40/IntelliJ IDEA.app/Contents/plugins/maven/lib/maven3`

This is an example how the `package` command is run:

```shell
/Library/Java/JavaVirtualMachines/jdk1.8.0_202.jdk/Contents/Home/bin/java -Dmaven.multiModuleProjectDirectory=/Users/varijkapil/Documents/Work/Development/JavaEE/work-service -Dmaven.home=/Applications/JetBrains/apps/IDEA-U/ch-0/211.7442.40/IntelliJ IDEA.app/Contents/plugins/maven/lib/maven3 -Dclassworlds.conf=/Applications/JetBrains/apps/IDEA-U/ch-0/211.7442.40/IntelliJ IDEA.app/Contents/plugins/maven/lib/maven3/bin/m2.conf -Dmaven.ext.class.path=/Applications/JetBrains/apps/IDEA-U/ch-0/211.7442.40/IntelliJ IDEA.app/Contents/plugins/maven/lib/maven-event-listener.jar -javaagent:/Applications/JetBrains/apps/IDEA-U/ch-0/211.7442.40/IntelliJ IDEA.app/Contents/lib/idea_rt.jar=54014:/Applications/JetBrains/apps/IDEA-U/ch-0/211.7442.40/IntelliJ IDEA.app/Contents/bin -Dfile.encoding=UTF-8 -classpath /Applications/JetBrains/apps/IDEA-U/ch-0/211.7442.40/IntelliJ IDEA.app/Contents/plugins/maven/lib/maven3/boot/plexus-classworlds.license:/Applications/JetBrains/apps/IDEA-U/ch-0/211.7442.40/IntelliJ IDEA.app/Contents/plugins/maven/lib/maven3/boot/plexus-classworlds-2.6.0.jar org.codehaus.classworlds.Launcher -Didea.version=2021.1.2 package
```

I see in `MavenEnvironment` that `maven.home` is given preference, but cannot figure out why it is not considered when this command is run? Instead it writes the following in the pre-commit bash script
`/Users/varijkapil/Documents/Work/Development/JavaEE/work-service/mvn`

Maybe a little help in this direction would be helpful.